### DISCUSSION TITLE
[Feature] Quickly hide/unhide External Libraries from Timeline

### DISCUSSION THREAD

**Discussion by BugsWorld** [immich-app/immich#6134] [Feature Request]
### The feature

A toggle that we can turn on to quickly hide all assets related to a specific External library. This is extremely useful when you wanna browse your timeline in different "modes". So for example you want to scroll through your timeline and see only your Family memories, so you hide all other libraries and keep the Family one only. Or you want to browse your personal pictures only through the timeline, so you only keep that one on. You get the idea, very useful, very neat. 

To take it a step further, it would be nice if one user could share a specific library with another user in a way that would add all the pictures to the user's timeline and not just under the "Sharing" tab. 

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by aisbergde** [comment]
It's a very good idea. Assets could be moved between libraries. It is on the road map to use fingerprints to detect, if it is the same asset or not. And if the source folder for the upload would be saved, then we could move for example screenshots into a separate library.

I want to extend the idea:
if we could assign custom tags and filter by them, then the external libraries are just special cases: each library can have its own tag, which is automatically assigned to all assets. Then I even don't need to move assets between libraries. We would only need a good system to filter by tags.

**Comment by gschintgen** [comment]
This feature would also be very useful for a mixed usecase in combination with e.g. Digikam:

- Use background backup for everyday operation and immediate cross-device availability.
- Periodically, e.g. after a vacation, transfer/manage/tag/edit/curate/etc in Digikam and the re-add the finalized collection in Immich as read-only external library.
- Remove the originals in Immich's upload library.

It's that last step that could be made more convenient by temporarily hiding external libraries: It'd be easier to see which assets are only the unmanaged backups in the default upload library.

I know that may not be the intended all-Immich workflow, but I'd guess that a sizable number of Immich users do try to make it work in combination with external management tools.

And there is of course the usecase given by OP which I fully agree with.

**Comment by ayalavalva** [comment]
This feature would be incredibly useful for filtering out external libraries containing photos and videos shared by others on social networks and chat apps. Currently, I’m unsure how to handle all the pictures from events I attended but don’t necessarily want mixed into my timeline alongside my own photos and videos.

**Comment by ty-han** [comment]
Count me in for hiding external libraries from the main timeline. I would prefer to access my external libraries in a separate view.

**Comment by problemss** [comment]
I like this idea and would like to add the ability to save that filter. For example I have two libraries, I want to have my app load one of the libraries by default and my wife’s app to load another by default. We can always toggle to the other or all but ideally the default view would be saved to our profile.

**Comment by StefanMihaylov** [comment]
This feature would be useful. In my case I have two external folders. The first folder contains hundreds of images from my DSLR camera, grouped into several albums (500+ images each), the second folders contains pics and videos from my phones. I'm looking for a feature that hides images from the timeline based on the external folder filter or configuration in the albums (hide/show in the timeline).

**Comment by matttarsi** [comment]
I would love the ability to select which libraries are currently visible in my timeline. Navidrome just recently implemented this feature and I think it works quite well. A toggle in the sidebar would be really nice.

<img width="488" height="414" alt="image" src="https://github.com/user-attachments/assets/b17697a6-7c7d-4306-9ca8-5e4a28f5f15b" />

**Comment by wsupnet** [comment]
+1 for this feature. My use case: I'm running Immich on a NAS as the single source of truth for my family's photos. I have my own photos as one external library and a family member's photos as a separate external library, each owned by the respective user.

Even with Partner Sharing's "Show in timeline" toggle, there's no equivalent control at the external library level. A simple per-library visibility toggle would make it much easier to switch between "my photos only" and "all libraries" views without having to archive assets or use workarounds like exclusion patterns. 

Ideally this would be a toggle visible in the user's own settings (not just admin), so each user can control which of their own external libraries appear in their timeline. Consistency with how Partner Sharing handles the "show in timeline" toggle would be a great model to follow. The above screenshot from @matttarsi would be a perfect example.

**Comment by abhinav4848** [comment]
The only way to do it now is to create a new user and create a new library specifically for that user. (add any paths for them as you want, it can be the same path I assume that some other user already has). Then proceed to log in as that user so only the library owned by them  (and whatever paths are included in that library) are visible in their timeline.
